src/js/angular/clustermanagement/services/cluster-context.service.js

```diff
@@ -64,6 +63,36 @@ function ClusterContextService(EventEmitterService) {
         emit(ClusterEventName.CLUSTER_VALID_UPDATED, isValid());
     };
 
+    /**
+     * Marks an item for deletion from the cluster.
+     * @param {Node|Location} itemToDelete - The item to delete. This can either be a Node or a Location object.
+     * @throws {Error} Throws an error if the item to delete is not provided or invalid.
+     * @return {void}
+     */
+    const deleteFromCluster = (itemToDelete) => {
+        if (!itemToDelete) {
+            throw new Error("Invalid node");
+        }
+        _clusterView.deleteFromCluster(itemToDelete);
+        emitUpdateClusterView();
+    };
+
+    /**
+     * Restores a previously deleted node back into the cluster.
+     * @param {Node} node - The node to restore.
+     * @throws {Error} Throws an error if the node is not provided or invalid.
+     * @return {void}
+     */
+    const restoreNode = (node) => {
+        if (!node) {
+            throw new Error("Invalid node");
+        }
+
+        const index = _clusterView.getDeleteFromCluster().findIndex((nodeToDelete) => nodeToDelete.endpoint === node.endpoint);
```

Comment:
Here again. Move this `findIndex...` in the model to prevent abstraction to leak here and there and to make the code harder to be extended in the future

Reply:
done